commands/audit/sca/gem/gem.go

```diff
@@ -0,0 +1,460 @@
+	if err != nil {
+		return "", fmt.Errorf("failed to execute 'ruby --version': %w", err)
+	}
+
+	versionStr := string(output)
+	fields := strings.Fields(versionStr)
+	if len(fields) < 2 {
+		return "", fmt.Errorf("unexpected ruby version output: %s", versionStr)
+	}
+	actualVersion := fields[1]
+
+	// Extract just major.minor from actual version and required version
+	actualMajor, actualMinor, err := parseMajorMinor(actualVersion)
+	if err != nil {
+		return "", err
+	}
+
+	requiredMajor, requiredMinor, err := parseMajorMinor(rubyV2)
+	if err != nil {
+		return "", err
+	}
+
+	if actualMajor < requiredMajor || (actualMajor == requiredMajor && actualMinor < requiredMinor) {
+		return "", fmt.Errorf(
+			"ruby dependency tree building requires ruby %s or higher; current version: %s",
+			rubyV2, actualVersion,
+		)
+	}
+
+	return bundleExecPath, nil
+}
+func getGemCmd(execPath, workingDir, cmd string, args ...string) *io.Command {
+	command := io.NewCommand(execPath, cmd, args)
+	command.Dir = workingDir
+	return command
+}
+
+// calculateDependencies orchestrates the generation and parsing of Gemfile.lock to build the dependency graph.
+// It first runs 'bundle lock' to ensure Gemfile.lock is up-to-date, then parses the lock file,
+// unmarshals it into an internal structure, and finally builds the Xray-compatible graph.
+// Returns:
+//
+//	dependencyTrees: A slice of top-level dependency nodes.
+//	uniqueDeps: A slice of unique dependency IDs found in the graph.
+func calculateDependencies(bundleExecPath, workingDir string, _ cliUtils.AuditParams) (dependencyTrees []*xrayUtils.GraphNode, uniqueDeps []string, err error) {
+	log.Debug("Ensuring Gemfile.lock is up to date using 'bundle lock'...")
+	if _, err = getGemCmd(bundleExecPath, workingDir, "lock").RunWithOutput(); err != nil {
+		err = fmt.Errorf("failed to execute 'bundle lock': %w. Ensure Gemfile is present and bundle can run", err)
+		return
+	}
+
+	lockFilePath := filepath.Join(workingDir, "Gemfile.lock")
+	if _, statErr := os.Stat(lockFilePath); os.IsNotExist(statErr) {
+		err = fmt.Errorf("gemfile.lock not found at '%s' after running 'bundle lock'", lockFilePath)
+		return
+	}
+
+	gemGraphInfoContent, err := parseGemfileLockDeps(lockFilePath)
+	if err != nil {
+		err = fmt.Errorf("error generating gem graph JSON from Gemfile.lock: %w", err)
+		return
+	}
+	var gemInput GemGraphInput
+	if err = json.Unmarshal(gemGraphInfoContent, &gemInput); err != nil {
+		err = fmt.Errorf("failed to unmarshal gem graph JSON: %w", err)
+		return
+	}
+	if len(gemInput.Graph.Nodes) == 0 {
+		log.Debug("No gem dependencies found in GemGraphInput after parsing Gemfile.lock.")
+		return []*xrayUtils.GraphNode{}, []string{}, nil
+	}
+
+	projectRootNode := buildFullGemDependencyGraph(gemInput)
+
+	if projectRootNode != nil && projectRootNode.Nodes != nil {
+		dependencyTrees = projectRootNode.Nodes
+	} else {
+		dependencyTrees = []*xrayUtils.GraphNode{}
+	}
```

Comment:
is there a case where `projectRootNode` is not null and `projectRootNode.Nodes` are? if so why aren't we adding `projectRootNode`?